erigon-lib/commitment/hex_patricia_hashed.go

```diff
@@ -2033,31 +2064,46 @@ func (hph *HexPatriciaHashed) Process(ctx context.Context, updates *Updates, log
 
 		default:
 		}
-
+		hph.metrics.Accounts.UpdatesInc(plainKey)
+		hph.metrics.Accounts.currentPlainKey = plainKey
+		hph.metrics.Accounts.UpdatesStorageInc(plainKey)
 		if hph.trace {
 			fmt.Printf("\n%d/%d) plainKey [%x] hashedKey [%x] currentKey [%x]\n", ki+1, updatesCount, plainKey, hashedKey, hph.currentKey[:hph.currentKeyLen])
 		}
 		// Keep folding until the currentKey is the prefix of the key we modify
 		for hph.needFolding(hashedKey) {
+			startFold := hph.metrics.Now()
+			hph.metrics.Accounts.FoldsInc(hph.metrics.Accounts.currentPlainKey)
 			if err := hph.fold(); err != nil {
 				return fmt.Errorf("fold: %w", err)
 			}
+			hph.metrics.TotalFoldingTimeInc(startFold)
+			hph.metrics.Accounts.TotalFoldingTimeInc(hph.metrics.Accounts.currentPlainKey, startFold)
 		}
 		// Now unfold until we step on an empty cell
 		for unfolding := hph.needUnfolding(hashedKey); unfolding > 0; unfolding = hph.needUnfolding(hashedKey) {
+			startUnfold := hph.metrics.Now()
+			hph.metrics.Accounts.UnfoldsInc(hph.metrics.Accounts.currentPlainKey)
+			hph.metrics.Unfolds.Add(1)
```

Comment:
i would replace this 3 lines here and 2 lines after with 
`unfoldDone := hph.metrics.AddUnfolding(plainKey)`
...
`unfoldDone()`

so unfoldDone, if metrics disabled, is noop just to avoid panic, and if metrics enabled, it keeps time record in closure as well as `plainKey`. That would be much easier to work with and disable if needed.